rust/sealevel/programs/ism/multisig-ism-message-id/src/processor.rs

```diff
@@ -249,7 +249,7 @@ fn verify(
                 merkle_tree_hook_address: metadata.origin_merkle_tree_hook,
                 mailbox_domain: message.origin,
                 root: metadata.merkle_root,
-                index: message.nonce,
+                index: metadata.merkle_index,
```

Comment:
maybe a good time to consider using multiple ISMs in e2e

Reply:
fwiw this is the only ISM impl we have on Sealevel - I think if we were sending EVM <> Sealevel in e2e we maybe would've caught this

Reply:
actually, since the merkle tree lives in the mailbox, I'm a bit unsure about how we could end up with a leaf index that's different from the message nonce, even EVM <> Sealevel

Reply:
For Sealevel origin we'd never get in this situation but for EVM -> Sealevel it's possible (which is what surfaced this)

Iiuc in our e2e tests when we send messages from the EVM we sometimes double insert or skip insertion into the merkle tree, so it'd test the case where nonce != merkle index